File: ruby/7kyu/dna.rb
# Complementary DNA
# https://www.codewars.com/kata/complementary-dna/ruby

def DNA_strand(dna)
  dna_replacement = { 'A' => 'T', 'T' => 'A', 'C' => 'G', 'G' => 'C' }
  dna.split('').map { |x| dna_replacement[x] }.join('')
end

File: javascript/7kyu/dna.js
// Complementary DNA
// https://www.codewars.com/kata/complementary-dna/javascript

function DNAStrand(dna) {
  const dnaReplacement = {
    A: 'T', T: 'A', C: 'G', G: 'C',
  };
  return dna
    .split('')
    .map((e) => dnaReplacement[e])
    .join('');
}
